### PR TITLE
Fix bugs related to journal replay

### PR DESCRIPTION
**Pull request by pleroy**
* Enter the out pointers in the pointer map.
* Remove the pointers from the map when they are deleted.

### CODE REVIEW

File: journal/profiles.cpp [hunk 1]
```diff
@@ -20,13 +20,23 @@ template<typename T>
 void Insert(not_null<Player::PointerMap*> const pointer_map,
             std::uint64_t const address,
             T* const pointer) {
-  auto const inserted_pointer = const_cast<std::remove_cv<T>::type*>(pointer);
+  void* const inserted_pointer =
+      static_cast<void*>(const_cast<std::remove_cv<T>::type*>(pointer));
```
Comment (eggrobin) [inline journal/profiles.cpp]: Do we still need the `const_cast` then?

File: journal/profiles.cpp [hunk 2]
```diff
@@ -596,6 +607,8 @@ void SetPlottingFrame::Run(Message const& message,
   auto* navigation_frame = DeserializePointer<NavigationFrame*>(
                                *pointer_map, in.navigation_frame());
   ksp_plugin::principia__SetPlottingFrame(plugin, &navigation_frame);
+  // TODO(phl): This may delete the previous plotting frame, which should thus
+  // be removed from |pointer_map|.
```
Comment (eggrobin) [inline journal/profiles.cpp]: No, the `pointer_map` doesn't represent the memory (we don't care about the memory), it represents the objects that the caller of the DLL owns directly. Thus `SetPlottingFrame` "deletes" (consumes) *its argument*.

File: journal/profiles.cpp (hunk 1 above)
Comment (pleroy) [inline journal/profiles.cpp]: Yes, of course.

File: journal/profiles.cpp (hunk 2 above)
Comment (pleroy) [inline journal/profiles.cpp]: Done.

**Pull request merged**